### ISSUE TITLE
Swipe back with empty history stack breaks navigation (iOS tested only)

### ISSUE THREAD

**Issue by ignaciofuentes** [NativeScript/NativeScript#2836]
Simply swiping back when the history stack is empty will break subsequent navigation attempts.
This applies to completely fresh stacks or stacks that get cleared by using clearHistory:true
Swiping back for a second time "fixes" it.

[Screencast here](http://screencast.com/t/dgUJlh9r3)

Simple repo
main-page.js
```
var frameModule = require("ui/frame");
var topmost = frameModule.topmost();

function onNavigatingTo(args) {
    var page = args.object;
}

function onTap(args) {
    var topmost = frameModule.topmost();
    topmost.navigate("second-page");
}
exports.onTap = onTap;
exports.onNavigatingTo = onNavigatingTo;
```

main-page.xml
```
<Page xmlns="http://schemas.nativescript.org/tns.xsd" 
navigatingTo="onNavigatingTo">
    <Page.actionBar>
      <ActionBar title="First Page"/>
  </Page.actionBar>  
  <StackLayout>
    <Button text="TAP" tap="onTap" />
  </StackLayout>
</Page>
```
second-page.xml

```
<Page xmlns="http://schemas.nativescript.org/tns.xsd">
  <Page.actionBar>
      <ActionBar title="Second Page"/>
  </Page.actionBar>
  <StackLayout>
    <Label class="title" text="Second page" />
  </StackLayout>
</Page>
```

**Comment by ignaciofuentes** [comment]
This is probably the root cause of:

[nativescript-angular/issues/486](https://github.com/NativeScript/nativescript-angular/issues/486)

**Comment by ignaciofuentes** [comment]
@vakrilov @NickIliev

**Comment by NickIliev** [comment]
Hey @ignaciofuentes - thank you for your help and screencast!

I have created the exact same project and with or without **_clearHitory_** i was 
 able to reproduce thie behavior you are experiencing. [Here is the sample project](https://github.com/NickIliev/NativeScript-issues/tree/master/NativeScript-Angular/issue_ng_486) I have created (tested with iOS 10 and ios 9.2 - created with 2.3.0 and also tested with modules@next) - on each occasion the exact same behavior is observed.. after the first swipe back and an atttemt to swipe back on a page with a cle4arHistory:true (or the main app page) the navigation got broken.
